polclassifier/ml_logic/models.py

```diff
@@ -62,3 +61,46 @@ def evaluate_model_svm(model, X, y):
     predictions = model.predict(X)
     accuracy = accuracy_score(y, predictions)
     return accuracy
+
+
+def randomized_search_model_knn(X, y):
+    # Define the hyperparameter grid
+    param_grid = {
+    'n_neighbors':  N_NEIGHBORS,  # Penalty parameter C (regularization parameter)
+    'leaf_size': LEAF_SIZE
+    }
+
+    # Create an KNN classifier
+    knn_classifier = KNeighborsClassifier()
+
+    # Perform random search cross-validation
+    random_search = RandomizedSearchCV(knn_classifier, param_distributions=param_grid, n_iter=10, scoring='accuracy', cv=5, n_jobs=-1)
```

Comment:
The n_iter should be way bigger, can you try something larger, seeing if you can get better performance? Here, you are only trying 10 combinations in total both for leaf_size and for how many closest neighbors to consider and since they are both set to anything between 1 and 100, that doesn't explore the option space that effectively. You could also consider doing multiple grid searches, where you specify what exact values to try . In the first run you could and guess what the "right range" likely is (so trying 5, 10, 20, 30, 40, 60, 80, 100 maybe) and in the next run zone in on one of those ranges (if for example 10 and 20 got better scores than the rest, then try every value from 5-25).